fix #639 use <details> for story previews

fedemp opened the pull request:
Use the `<details>` element when showing story previews, so browsers will show matches inside when using the native search in page feature.

<!--
I (@pushcx) try to timebox non-urgent Lobsters maintenance to the scheduled office hours streams (https://push.cx/stream), so it may take me a couple days to respond. If you don't want your issue or PR reviewed on a stream, say so and I won't.

If your PR is part of your classwork as a student, please explain what your assignment is. It helps me give you better feedback.

Do not submit code written by LLM-powered coding tools because of the uncertainty around their output's copyright: 
https://en.wikipedia.org/wiki/Artificial_intelligence_and_copyright
-->

fedemp commented:
I'm open to suggestions. As mentioned in the original issue, the browser may find text that is hidden due to `overflow: hidden`. `<details>` automatically opens when that happens.

The downside is that the first characters of the description are duplicated inside the `<summary>`. I lessen that by changing the color and font size of the summary when details is open. 

<img width="1778" height="349" alt="image" src="https://github.com/user-attachments/assets/868c5e2c-55ae-441a-be93-39259c8677f7" />

<img width="1778" height="642" alt="image" src="https://github.com/user-attachments/assets/647f1a2b-5d66-476d-953d-f7ed8111bbb3" />

pushcx commented on db/queue_schema.rb:
```diff
@@ -10,132 +10,5 @@
-  add_foreign_key "solid_queue_ready_executions", "solid_queue_jobs", column: "job_id", on_delete: :cascade
-  add_foreign_key "solid_queue_recurring_executions", "solid_queue_jobs", column: "job_id", on_delete: :cascade
-  add_foreign_key "solid_queue_scheduled_executions", "solid_queue_jobs", column: "job_id", on_delete: :cascade
+ActiveRecord::Schema[8.0].define(version: 0) do
```
Looks like a hassle in your dev env; the queue didn't get set up correctly and so Rails wiped out the schema. Please revert changes to this file.

pushcx commented on db/cache_schema.rb:
```diff
@@ -10,15 +10,5 @@
-    t.index ["key_hash", "byte_size"], name: "index_solid_cache_entries_on_key_hash_and_byte_size"
-    t.index ["key_hash"], name: "index_solid_cache_entries_on_key_hash", unique: true
-  end
+ActiveRecord::Schema[8.0].define(version: 0) do
```
Revert.

fedemp commented:
I have added a rule so the `summary` is hidden when `details` is open, without sacrificing the ability to toggle it. I'm just concerned how screen readers will react to the `summary` suddenly disappearing; we will have to wait for feedback. :shrug: 

The black border is the focus ring, not something that I've added.

<img width="797" height="368" alt="image" src="https://github.com/user-attachments/assets/59ffa670-c1e0-4748-9dec-b4d4f0731b0a" />

fedemp commented on app/views/stories/_listdetail.html.erb:
```diff
@@ -53,10 +53,11 @@ class="story <%= story.current_upvoted? ? "upvoted" : "" %>
     <% if @user&.show_story_previews? %>
-      <% if (sc = story.description_or_story_text(500)).present? %>
-        <div class="story_content">
+      <% if (sc = story.description_or_story_text()).present? %>
```
There are no story texts in `singledetails`. And to be honest, it actually calls my attention that every story *in the dev environment* has a story text but when I visit the single details page, there's nothing there.
